### DISCUSSION TITLE
Write rego policy for optional attributes

### DISCUSSION THREAD

**Discussion by vkosuri** [open-policy-agent/community#531] [Community]
The data is anonymized, and the example playground is here https://play.openpolicyagent.org/p/F5jzxTd173

In the below JSON body, the attributes ``columns`` and ``rows`` may present and may not present in the json body(optional attributes, sometimes one attribute present, both attributes present, or none).

We need to validate those values if they present with exact text, the user may input the wrong values where we need to restrict API and not push the stack. Any suggestion how to handle this case?

```JSON
{
    "cSide": {
        "Side": "default",
        "test": {
            "columns": "123",
            "rows": "DEFAULT",
            "table": "DEFAULT",
            "testVersions": {
                "enable1": true,
                "enable2": true,
                "enable3": true
            }
        }
    },
    "name": "table"
}
```

**Comment by charlieegan3** [comment]
How about something like this:

```rego
package example

import future.keywords.contains
import future.keywords.if

default allow := false

allow := deny == set()

deny contains msg if {
	input.cSide.test.rows
	input.cSide.test.rows != "DEFAULT"

	msg := "rows was not the set to DEFAULT"
}

deny contains msg if {
	input.cSide.test.columns
	input.cSide.test.columns != "DEFAULT"

	msg := "columns was not the set to DEFAULT"
}
```

https://play.openpolicyagent.org/p/F5jzxTd173

**Comment by ratnapolepalli** [comment]
IS there a way to integrate the below functions
```shell
validate_rows(rows) = false {
	rows != "DEFAULT"
}

validate_rows(rows) {
	rows == "DEFAULT"
}

validate_columns(columns) = false {
	columns != "DEFAULT"
}

validate_columns(columns) {
	columns == "DEFAULT"
}
```

**Reply by charlieegan3** [reply]
Perhaps like this? https://play.openpolicyagent.org/p/sqQ8wwgGDU
